lib/jwe/decrypt.js

```diff
@@ -223,12 +223,14 @@ function JWEDecrypter(ks, globalOpts) {
       promise = promise.then(function(jwe) {
         var processing = [];
         handlerKeys.forEach(function(h) {
-          h = extraHandlers[h];
+          var hv = extraHandlers[h];
           var p;
-          if ("function" === typeof h) {
-            p = h(jwe);
-          } else if ("object" === typeof h && "function" === typeof h.prepare) {
-            p = h.prepare(jwe);
+          if ("function" === typeof hv) {
+            p = hv(jwe);
+          } else if ("object" === typeof hv && "function" === typeof hv.prepare) {
+            p = hv.prepare(jwe);
+        } else if ("boolean" === typeof hv && hv !== jwe.header.hasOwnProperty(h)) {
+              p = Promise.reject(new Error("bad crit header"))
```

Comment:
One could argue that is better suing full words instead of `crit`.

Reply:
Indeed, one could argue that. I do argue that a good argument includes concrete suggestions for improvement. ;) I simply stuck with the wording in the doc, which states “crit header” the last time I checked. Of course, one could argue that the docs could be improved as well. But this is beyond the scope of this PR and not my argument.

Reply:
Sure, I didnt meant to say you didnt do a great job, because I believe thats the case here. I just tried to see if there is space for improvement. Excuse me you are right this is how it is being called `crit`

Reply:
Cheers, I am fine with improvement suggestions but I prefer them to be specific rather than generic :)